packages/snap/src/utils/validation.ts

```diff
@@ -0,0 +1,91 @@
+import type { Hex } from '@metamask/utils';
+import { isValidHexAddress } from '@metamask/utils';
+import { ethers } from 'ethers';
+import { assert, define, object, optional, StructError } from 'superstruct';
+
+import { throwError } from './util';
+import { CONFIG_ERROR_MESSAGES, CONFIG_KEYS } from '../constants/chainConfig';
+import type { ChainConfig } from '../keyring';
+
+const EthereumAddress = define(
+  'EthereumAddress',
+  (value) => typeof value === 'string' && isValidHexAddress(value as Hex),
+);
+
+const Url = define('Url', (value) => {
+  const urlPattern =
+    /^(https?:\/\/)?[\w.-]+(:\d{2,6})?(\/[\w.-]*)?(\?[\w.&=%-]*)?$/u;
+  return typeof value === 'string' && urlPattern.test(value);
+});
+
+const PrivateKey = define('PrivateKey', (value) => {
+  return typeof value === 'string' && ethers.isHexString(value);
+});
+
+const ChainConfigStruct = object({
+  [CONFIG_KEYS.SIMPLE_ACCOUNT_FACTORY]: optional(EthereumAddress),
```

Comment:
We should use `exactOptional` here from `keyring-api`, using optional can lead to the failure of the state to be saved in the case were it is undefined.

Reply:
Addressed here [39e96b5](https://github.com/MetaMask/snap-account-abstraction-keyring/pull/20/commits/39e96b574777574fd43804129565fcdea3be01a9)